Mage.Sets/src/mage/cards/d/DannyPink.java

```diff
@@ -0,0 +1,58 @@
+
+package mage.cards.d;
+
+import java.util.UUID;
+
+import mage.MageInt;
+import mage.abilities.TriggeredAbility;
+import mage.abilities.common.OneOrMoreCountersAddedTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.continuous.GainAbilityControlledEffect;
+import mage.abilities.keyword.MentorAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.filter.StaticFilters;
+
+/**
+ *
+ * @author Leonan
+ */
+public final class DannyPink extends CardImpl {
+
+    public DannyPink(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{U}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.subtype.add(SubType.ADVISOR);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(3);
+
+        // Mentor
+        this.addAbility(new MentorAbility());
+
+        // Creatures you control have “Whenever one or more counters are put on this creature for the first time each turn, draw a card.”
+        TriggeredAbility gainedAbility = new OneOrMoreCountersAddedTriggeredAbility(
```

Comment:
You need to redo this ability to likely be a custom TriggeredAbility based off [Axgard Artisan's](https://github.com/magefree/mage/blob/master/Mage.Sets/src/mage/cards/a/AxgardArtisan.java) that works correctly so that each creature only triggers the first time counters are put on it each turn.

Reply:
this.setTriggerPhrase("Whenever one or more +1/+1 counters are put on {this} for the first time each turn, ");
Here, instead of {this}, what do I put?

Reply:
"OneOrMoreCountersAddedTriggeredAbility" triggers every time counters are put on the creature. You need a different ability - likely similar or identical to [AxgardArtisanTriggeredAbility](https://github.com/magefree/mage/blob/master/Mage.Sets/src/mage/cards/a/AxgardArtisan.java#L47-L77) - which triggers when one or more counters are added, but only the first time that happens each turn.

Reply:
It may even be a little more complicated than AxgardArtisan because if the other creature gained a counter prior to Danny Pink arriving onfield, it still shouldn't trigger.

I found [Botanical Brawler](https://github.com/magefree/mage/blob/master/Mage.Sets/src/mage/cards/b/BotanicalBrawler.java#L56-L98) which should work the same, but I haven't checked whether it works correctly.

Note: I am not the most savvy with card implementations, so other peoples coments should have more weight than mine.

Reply:
You would need to actually do a different watcher and some different logic - Axgard Artisan (and Botanical Brawler) are exclusively +1/+1 counters, where Danny Pink is any counter; however, I believe that setup of watcher should work correctly where it will observe counters added before Danny Pink enters the battlefield.

Reply:
(These added complications and custom TriggeredAbility/Watcher are the reason why nobody has previously implemented Danny Pink.)